sqsapp.java

```diff
@@ -0,0 +1,146 @@
+
+        // Create a fifo queue
+
+        Map<String, String> queueAttributes = new HashMap<String, String>();
+        queueAttributes.put("FifoQueue", "true");
+        queueAttributes.put("ContentBasedDeduplication", "true");
+
+        CreateQueueRequest createFifoQueueRequest = new CreateQueueRequest("baeldung-queue.fifo").withAttributes(queueAttributes);
+        String fifoQueueUrl = sqs.createQueue(createFifoQueueRequest)
+            .getQueueUrl();
+
+        System.out.println(fifoQueueUrl);
+
+        // Set up a dead letter queue
+
+        String deadLetterQueueUrl = sqs.createQueue("baeldung-dead-letter-queue")
+            .getQueueUrl();
+
+        GetQueueAttributesResult deadLetterQueueAttributes = sqs.getQueueAttributes(new GetQueueAttributesRequest(deadLetterQueueUrl).withAttributeNames("QueueArn"));
+
+        String deadLetterQueueARN = deadLetterQueueAttributes.getAttributes()
+            .get("QueueArn");
+
+        SetQueueAttributesRequest queueAttributesRequest = new SetQueueAttributesRequest().withQueueUrl(standardQueueUrl)
+            .addAttributesEntry("RedrivePolicy", "{\"maxReceiveCount\":\"2\", " + "\"deadLetterTargetArn\":\"" + deadLetterQueueARN + "\"}");
+
+        sqs.setQueueAttributes(queueAttributesRequest);
+
+        // Send a message to a standard queue
+
+        Map<String, MessageAttributeValue> messageAttributes = new HashMap<>();
+
+        messageAttributes.put("AttributeOne", new MessageAttributeValue().withStringValue("This is an attribute")
+            .withDataType("String"));
+
+        SendMessageRequest sendMessageStandardQueue = new SendMessageRequest().withQueueUrl(standardQueueUrl)
+            .withMessageBody("A simple message.")
+            .withDelaySeconds(30) // Message will arrive in the queue after 30 seconds. We can use this only in standard queues
+            .withMessageAttributes(messageAttributes);
+
+        sqs.sendMessage(sendMessageStandardQueue);
+
+        // Send a message to a fifo queue
+
+        SendMessageRequest sendMessageFifoQueue = new SendMessageRequest().withQueueUrl(fifoQueueUrl)
+            .withMessageBody("FIFO Queue")
+            .withMessageGroupId("baeldung-group-1")
+            .withMessageAttributes(messageAttributes);
+
+        sqs.sendMessage(sendMessageFifoQueue);
+
+        // Send multiple messages
+
+        List<SendMessageBatchRequestEntry> messageEntries = new ArrayList<>();
+        messageEntries.add(new SendMessageBatchRequestEntry().withId("id-1")
+            .withMessageBody("batch-1")
+            .withMessageGroupId("baeldung-group-1"));
+        messageEntries.add(new SendMessageBatchRequestEntry().withId("id-2")
+            .withMessageBody("batch-2")
+            .withMessageGroupId("baeldung-group-1"));
+
+        SendMessageBatchRequest sendMessageBatchRequest = new SendMessageBatchRequest(fifoQueueUrl, messageEntries);
+        sqs.sendMessageBatch(sendMessageBatchRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code makes a call to a batch operation, but it does not handle any errors returned by that batch operation. Consider adding error checks and verifications.